worldedit-core/src/main/java/com/sk89q/worldedit/entity/Player.java

```diff
@@ -437,14 +437,16 @@ default void unregister() {
                     } else {
                         continue;
                     }
-                    WorldEdit.getInstance().getExecutorService().submit(() -> {
+                    Fawe.instance().getClipboardExecutor().submit(getUniqueId(), () -> {
                         doc.close(); // Ensure closed before deletion
                         doc.getFile().delete();
                     });
                 }
             }
-        } else if (Settings.settings().CLIPBOARD.DELETE_ON_LOGOUT || Settings.settings().CLIPBOARD.USE_DISK) {
-            WorldEdit.getInstance().getExecutorService().submit(() -> session.setClipboard(null));
+        } else if (Settings.settings().CLIPBOARD.USE_DISK) {
```

Comment:
I guess this is fine for now as it was like that before, but actually the clipboard can be set by plugins to arbitrary implementations, so relying on the settings isn't exactly correct.